tests/ui/pattern/deref-patterns/deref-box.rs

```diff
@@ -0,0 +1,29 @@
+//@ run-pass
+//! Deref patterns on boxes are lowered using built-in derefs, rather than generic `Deref::deref`
+//! and `DerefMut::deref_mut`. Test that they work as expected.
+
+#![feature(deref_patterns)]
+#![expect(incomplete_features)]
+
+fn unbox_1<T>(b: Box<T>) -> T {
+    let deref!(x) = b else { unreachable!() };
+    x
+}
+
+fn unbox_2<T>(b: Box<(T,)>) -> T {
+    let (x,) = b else { unreachable!() };
+    x
+}
+
```

Comment:
Let's also test partial moves:
```suggestion

fn unbox_separately<T>(b: Box<(T, T)>) -> (T, T) {
    let (x, _) = b else { unreachable!() };
    let (_, y) = b else { unreachable!() };
    (x, y)
}
```